pythonz/define.py

```diff
@@ -4,10 +4,23 @@
 # pythonz installer root path
 INSTALLER_ROOT = os.path.dirname(os.path.abspath(__file__))
 
+HOME = os.environ.get('HOME')
+SYSTEMWIDE_PATH = '/usr/local/pythonz'
+
 # Root
 # pythonz root path
-SYSTEMWIDE_PATH = '/usr/local/pythonz'
-ROOT = os.environ.get('PYTHONZ_ROOT') or os.path.join(os.environ['HOME'], '.pythonz') if not os.path.abspath(os.path.dirname(__file__)).startswith(SYSTEMWIDE_PATH) else SYSTEMWIDE_PATH
+PYTHONZ_ROOT = os.environ.get('PYTHONZ_ROOT')
+if   ( PYTHONZ_ROOT
+       and os.path.isdir(PYTHONZ_ROOT) ):
+    ROOT = PYTHONZ_ROOT
+elif ( HOME
+       and os.path.isdir(os.path.join(HOME, '.pythonz')) ):
+    ROOT = os.path.join(HOME, '.pythonz')
+elif os.path.isdir(SYSTEMWIDE_PATH):
```

Comment:
IIRC, when installed system wide by root but running it as a user none of
the env variables are set yet.
On May 28, 2015 1:50 PM, "Ian Buchanan" <notifications@github.com> wrote:

> In pythonz/define.py
> <https://github.com/saghul/pythonz/pull/82#discussion_r31225155>:
>
> >  # Root
> >  # pythonz root path
> > -SYSTEMWIDE_PATH = '/usr/local/pythonz'
> > -ROOT = os.environ.get('PYTHONZ_ROOT') or os.path.join(os.environ['HOME'], '.pythonz') if not os.path.abspath(os.path.dirname(__file__)).startswith(SYSTEMWIDE_PATH) else SYSTEMWIDE_PATH
> > +PYTHONZ_ROOT = os.environ.get('PYTHONZ_ROOT')
> > +if   ( PYTHONZ_ROOT
> > +       and os.path.isdir(PYTHONZ_ROOT) ):
> > +    ROOT = PYTHONZ_ROOT
> > +elif ( HOME
> > +       and os.path.isdir(os.path.join(HOME, '.pythonz')) ):
> > +    ROOT = os.path.join(HOME, '.pythonz')
> > +elif os.path.isdir(SYSTEMWIDE_PATH):
>
> I see the check for the startswith in the original but the logic escapes
> me. Which is why I unpacked the if statement, then propagated the structure
> into all these different places. Would you be so kind to explain it?
> Specifically, why does it matter if define.py is inside /usr/local/pythonz
> ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/saghul/pythonz/pull/82/files#r31225155>.
>